contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1845,7 +1861,8 @@ select_common_type_setop(ParseState *pstate, List *exprs, Node **which_expr)
 			continue;
 		else if (is_tsql_str_const(expr))
 			type = common_utility_plugin_ptr->lookup_tsql_datatype_oid("varchar");
-		else if (!is_tsql_char_type_with_len(type))
+		else if ((!isCase && !is_tsql_char_type_with_len(type)) ||
+		         (isCase && !is_tsql_char_type_with_len_case_exp(type)))
```

Comment:
Instead of creating new function `is_tsql_char_type_with_len_case_exp` and invoking it separately, I would suggest to extend the definition of `is_tsql_char_type_with_len` by passing extra arg so that we don't make multiple calls on functions like `is_tsql_nvarchar_datatype`.

Reply:
is_tsql_char_type_with_len   if we change the function defination for this we need to change it at other place(used in this function also tsql_select_common_typmod_hook) also Where we don't have context.

Reply:
Made the chnage just kept one function to reduce the redundant code.